core/src/test/java/org/apache/iceberg/rest/TestHTTPClient.java

```diff
@@ -133,6 +136,49 @@ public void testDynamicHttpRequestInterceptorLoading() {
     assertThat(((TestHttpRequestInterceptor) interceptor).properties).isEqualTo(properties);
   }
 
+  @Test
+  public void testHttpClientGetConnectionConfig() {
+    long connectionTimeout = 10L;
+    int socketTimeout = 10;
+    Map<String, String> properties =
+        ImmutableMap.of(
+            HTTPClient.REST_CONNECTION_TIMEOUT, String.valueOf(connectionTimeout),
```

Comment:
there should probably also be a test where only one of these is set. Also please add some (separate) tests where connection/socket timeout are being set to some random string